clang/test/Driver/sycl-ftarget-register-alloc-mode-warning.cpp

```diff
@@ -0,0 +1,25 @@
+// Test warning SYCL -ftarget-register-alloc-mode for pvc
+
+// RUN: %clang -fsycl -Xs "-device pvc -ze-opt-large-register-file" -### %s 2>&1 | FileCheck -check-prefix=LARGE %s
+// RUN: %clang -fsycl -Xs "-device pvc -ze-intel-128-GRF-per-thread" -### %s 2>&1 | FileCheck -check-prefix=SMALL %s
+// RUN: %clang -fsycl -Xs "-ze-intel-enable-auto-large-GRF-mode -device pvc" -### %s 2>&1 | FileCheck -check-prefix=AUTO %s
+
+// RUN: %clang -fsycl -fsycl-targets=intel_gpu_pvc -Xs "-ze-opt-large-register-file" -### %s 2>&1 | FileCheck -check-prefix=LARGE %s
```

Comment:
Did I miss any important cases here?

Reply:
So do we care about potential usages like `-Xs -ze-opt-large-register-file -Xs "-device pvc"`?

Reply:
im gonna go with no because i've never seen anybody do that and it seems like it would be pretty annoying to deal with